testnet/src/operations.rs

```diff
@@ -0,0 +1,185 @@
+use std::io::{Read, Write};
+
+use stacks::burnchains::{BurnchainSigner, PrivateKey};
+use stacks::chainstate::burn::{BlockHeaderHash, ConsensusHash, Opcodes, VRFSeed};
+use stacks::chainstate::stacks::StacksAddress;
+use stacks::net::StacksMessageCodec;
+use stacks::net::codec::{write_next};
+use stacks::net::Error as net_error;
+use stacks::util::hash::Hash160;
+use stacks::util::vrf::VRFPublicKey;
+use stacks::util::secp256k1::{MessageSignature, Secp256k1PublicKey, Secp256k1PrivateKey};
+
+
+#[derive(Debug, Clone)]
+pub enum BurnchainOperationType {
+    LeaderBlockCommit(LeaderBlockCommitPayload),
+    LeaderKeyRegister(LeaderKeyRegisterPayload),
+    UserBurnSupport(UserBurnSupportPayload)
+}
+
+
+#[derive(Debug, PartialEq, Clone, Eq)]
```

Comment:
My earlier comment got removed, somehow.  Please do not duplicate the burnchain operation implementation here, since it may be subject to change (especially when PoX is implemented).  Let's try and keep their definitions in one place: under `src/chainstate/burn/operations`.  Instead, please implement `StacksMessageCodec` for each Blockstack operation type.  The extra fields -- `txid`, `vtxindex`, `burn_header_hash`, and `block_height` -- should either be filled in from Bitcoin data, or supplied via the mocknet controller (which it already appears to be doing, except in a round-about way that involves converting a `BurnchainOperationType` into a `BlockstackOperationType`).

Reply:
You earlier comment is still here - https://github.com/blockstack/stacks-blockchain/pull/1381#discussion_r401850204.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1381/commits/eb79e9ca7a2a4dcd40ca61f61adecf054a45e41a

Reply:
Thanks :)